### PR TITLE
Exercise 4 - Login System

### PR DESCRIPTION
**Pull request by janrebpol**
Login System which user can input their username and password. The system has hardcoded username and password. If the inputs are correct it will display successful and if not it will display incorrect info.

### CODE REVIEW

File: 4_exe/LoginController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+include 'LoginModel.php';
```
Comment (hipe-francis) [inline 4_exe/LoginController.php]: Use namespace for your imports.

File: 4_exe/LoginController.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+class LoginController {
+    private $userModel;
+
+public function __construct() {
+    $this->userModel = new User();
+}
+
+public function verifyLogin() {
```
Comment (hipe-francis) [inline 4_exe/LoginController.php]: Format to proper indention.

File: 4_exe/LoginController.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+private function isPostRequest() {
+    return $_SERVER['REQUEST_METHOD'] === 'POST';
+}
+
+private function getPostData($key) {
+    return isset($_POST[$key]) ? $_POST[$key] : null;
+}
+}
```
Comment (hipe-francis) [inline 4_exe/LoginController.php]: All PHP files MUST end with a non-blank line, terminated with a single LF.

File: 4_exe/LoginModel.php [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?php 
+
+class User {
+    private $username = 'user';
+    private $password = 'password123';
+
+    public function validateInfo($infoUsername, $infoPassword) {
+        return ($infoUsername == $this->username) && ($infoPassword == $this->password);
+}
+}
```
Comment (hipe-francis) [inline 4_exe/LoginModel.php]: All PHP files MUST end with a non-blank line, terminated with a single LF.

File: 4_exe/LoginModel.php [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?php 
+
+class User {
+    private $username = 'user';
+    private $password = 'password123';
+
+    public function validateInfo($infoUsername, $infoPassword) {
+        return ($infoUsername == $this->username) && ($infoPassword == $this->password);
+}
```
Comment (hipe-francis) [inline 4_exe/LoginModel.php]: Format to proper indention.

File: 4_exe/LoginController.php [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+public function verifyLogin() {
+    $result = $this->initializeDefaultResult();
+
+    if ($this->isPostRequest()) {
+        $username = $this->getPostData('username');
+        $password = $this->getPostData('password');
+
+        if ($this->userModel->validateInfo($username, $password)) {
+            $result['success'] = true;
```
Comment (hipe-francis) [inline 4_exe/LoginController.php]: Use early return pattern.

File: 4_exe/login.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    if ($loginResult['success']) {
+        echo "<div class='message'>Login successful!</div>";
+    } else {
+        echo "<div class='message'>{$loginResult['message']}</div>";
+    }
+    ?>
+        <header>
+                <h1 class="masked-text">Login</h1>
+            </header>
+            <form action="" method="post">
```
Comment (hipe-francis) [inline 4_exe/login.php]: Does this work? Did you test this?

**Review by hipe-francis — COMMENTED** [review]
Please refer to my comments to this or your previous PR's for similar changes.

File: 4_exe/login.php (hunk 7 above)
Comment (janrebpol) [reply 4_exe/login.php]: yes sir, it works.

**Review by hipe-francis — APPROVED** [review]
LGTM 🚀

**Pull request merged**